tests/components/dyson/test_climate.py

```diff
@@ -154,236 +136,259 @@ def _get_device_heat_on():
+    await async_setup_component(hass, DYSON_DOMAIN, _get_config())
+    await hass.async_block_till_done()
+
+    device = mocked_devices.return_value[0]
+    device.temp_unit = TEMP_CELSIUS
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: "climate.temp_name", ATTR_FAN_MODE: FAN_FOCUS},
+        True,
+    )
+
+    set_config = device.set_configuration
+    assert set_config.call_args == call(focus_mode=FocusMode.FOCUS_ON)
+
+    await hass.services.async_call(
+        DOMAIN,
+        SERVICE_SET_FAN_MODE,
+        {ATTR_ENTITY_ID: "climate.temp_name", ATTR_FAN_MODE: FAN_DIFFUSE},
+        True,
+    )
+
+    set_config = device.set_configuration
+    assert set_config.call_args == call(focus_mode=FocusMode.FOCUS_OFF)
+
+
+@patch(
+    "homeassistant.components.dyson.DysonAccount.devices",
+    return_value=[_get_device_heat_on()],
+)
+@patch("homeassistant.components.dyson.DysonAccount.login", return_value=True)
+async def test_pure_hot_cool_link_state(mocked_login, mocked_devices, hass):
+    """Test set climate temperature."""
+    await async_setup_component(hass, DYSON_DOMAIN, _get_config())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes[ATTR_SUPPORTED_FEATURES] == SUPPORT_FLAGS
+    assert state.attributes[ATTR_TEMPERATURE] == 23
+    assert state.attributes[ATTR_CURRENT_TEMPERATURE] == 289 - 273
+    assert state.attributes[ATTR_CURRENT_HUMIDITY] == 53
+    assert state.state == HVAC_MODE_HEAT
+    assert len(state.attributes[ATTR_HVAC_MODES]) == 2
+    assert HVAC_MODE_HEAT in state.attributes[ATTR_HVAC_MODES]
+    assert HVAC_MODE_COOL in state.attributes[ATTR_HVAC_MODES]
+    assert len(state.attributes[ATTR_FAN_MODES]) == 2
+    assert FAN_FOCUS in state.attributes[ATTR_FAN_MODES]
+    assert FAN_DIFFUSE in state.attributes[ATTR_FAN_MODES]
+
+    device = mocked_devices.return_value[0]
+    update_callback = device.add_message_listener.call_args[0][0]
+
+    device.state.focus_mode = FocusMode.FOCUS_ON.value
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes[ATTR_FAN_MODE] == FAN_FOCUS
+
+    device.state.focus_mode = FocusMode.FOCUS_OFF.value
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.attributes[ATTR_FAN_MODE] == FAN_DIFFUSE
+
+    device.state.heat_mode = HeatMode.HEAT_ON.value
+    device.state.heat_state = HeatState.HEAT_STATE_OFF.value
+    await hass.async_add_executor_job(update_callback, MockDysonState())
+    await hass.async_block_till_done()
+
+    state = hass.states.get("climate.temp_name")
+    assert state.state == HVAC_MODE_HEAT
+    assert state.attributes[ATTR_HVAC_ACTION] == CURRENT_HVAC_IDLE
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6184085)